src/Aspire.Cli/DotNet/DotNetSdkInstaller.cs

```diff
@@ -130,6 +248,39 @@ private static string GetCurrentArchitecture()
         };
     }
 
+    /// <summary>
+    /// Gets the directory where .NET runtimes are stored.
+    /// </summary>
+    /// <returns>The full path to the runtimes directory.</returns>
+    private string GetRuntimesDirectory()
+    {
+        return executionContext.RuntimesDirectory.FullName;
+    }
+
+    /// <summary>
+    /// Gets the install script information based on the current platform.
+    /// </summary>
+    /// <returns>A tuple containing the script URL, script file name, and script runner command.</returns>
+    private static (string ScriptUrl, string ScriptFileName, string ScriptRunner) GetInstallScriptInfo()
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return (
+                "https://dot.net/v1/dotnet-install.ps1",
+                "dotnet-install.ps1",
+                "powershell"
```

Comment:
Some systems don't have `powershell` but only `pwsh`. In crank I try both with `--version` to ensure which one is available.